pkg/handlers/pdf.go

```diff
@@ -2,96 +2,54 @@ package handlers
 
 import (
 	"bytes"
+	"context"
 	"encoding/json"
-	"fmt"
 	"io"
 	"io/ioutil"
-	"mime/multipart"
 	"net/http"
-	"net/textproto"
-	"os"
 )
 
+type invokeLambdaFunc func(cxt context.Context, html string) ([]byte, error)
+
 // PDFHandler handles PDFs
 type PDFHandler struct {
 	HandlerBase
+	invokeLambda invokeLambdaFunc
 }
 
 // NewPDFHandler returns a new PDFHandler
-func NewPDFHandler() *PDFHandler {
-	return &PDFHandler{}
+func NewPDFHandler(invoker invokeLambdaFunc) *PDFHandler {
+	return &PDFHandler{invokeLambda: invoker}
+}
+
+type requestPayload struct {
+	HTML string `json:"html"`
 }
 
 // Handle returns an http.HandlerFunc
 func (h PDFHandler) Handle() http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		type GenerateRequest struct {
-			HTML     string `json:"html"`
-			Filename string `json:"filename"`
-		}
-
-		var data GenerateRequest
 
 		body, readErr := ioutil.ReadAll(r.Body)
 		if readErr != nil {
 			h.WriteErrorResponse(r.Context(), w, readErr)
 			return
 		}
 
-		parseErr := json.Unmarshal(body, &data)
+		var generateRequest requestPayload
```

Comment:
Non-blocking suggestion/consideration: You could possibly just do this as an inline anonymous struct, if you don't ever need to reference the `requestPayload` struct elsewhere, e.g.: 
```
generateRequest := struct{ 
    HTML string `json:"html"`
}{}
```

Reply:
Solid tip! I always forget that Go has those.

I'll try this out when I make the next change to this code.